dotCMS/src/main/java/com/dotmarketing/db/DbConnectionFactory.java

```diff
@@ -100,20 +112,34 @@ public static DataSource getDataSource() {
             synchronized (DbConnectionFactory.class) {
 
                 if (null == defaultDataSource) {
+                    if (DBPropertiesDatasourceStrategy.getInstance().existsDBPropertiesFile()) {
+                        defaultDataSource = DBPropertiesDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from db.properties file");
+                    } else if (System.getenv("connection_db_base_url") != null) {
+                        defaultDataSource = SystemEnvDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from system environment");
+                    } else {
+                        defaultDataSource = DockerSecretDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from Docker Secret");
+                    }
+
+                    if (null == defaultDataSource){
+                        defaultDataSource = TomcatDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from context.xml");
+                    }
 
                     try {
-                        final InitialContext ctx = new InitialContext();
-                        defaultDataSource = (DataSource) JNDIUtil.lookup(ctx, Constants.DATABASE_DEFAULT_DATASOURCE);
-                    } catch (Throwable e) {
-                        Logger.error(DbConnectionFactory.class,
-                                "---------- DBConnectionFactory: error getting dbconnection " + Constants.DATABASE_DEFAULT_DATASOURCE,
-                                e);
-                        if(Config.getBooleanProperty("SYSTEM_EXIT_ON_STARTUP_FAILURE", true)){
-                          e.printStackTrace();
-                          System.exit(1);
+                        //Adds datasource to JNDI if needed
```

Comment:
I think it should be a private method: addJNDIIfNeeded,  and remove the comment

Reply:
Done